plugins/transport/socket/main.go

```diff
@@ -87,18 +91,33 @@ func (s *Socket) Run(ctx context.Context, w transport.WriteFn, done chan bool) {
 	}
 	skt.Close()
 
-	s.logger.Infof("socket %s bind successful", laddr.Name)
-	go func() {
+	s.logger.Infof("socket listening on %s", laddr.Name)
+	go func(initBuffSize, maxBuffSize int64) {
+		msgBuffer := make([]byte, initBuffSize)
 		for {
-			var buff bytes.Buffer
-			n, err := io.Copy(&buff, pc)
-			if err != nil {
-				s.logger.Errorf(err, "failed reading message")
-			} else {
-				s.logger.Debugf("read message of size %d bytes", n)
+			n, err := pc.Read(msgBuffer)
+			if err != nil || n < 1 {
+				if err != nil {
+					s.logger.Errorf(err, "reading from socket failed")
+				}
+				done <- true
+				return
 			}
 			msg := buff.Bytes()
 
+			// whole buffer was used, so we are potentially handling larger message
+			// and have to increase buffer size
+			if n == len(msgBuffer) {
```

Comment:
Will this drop one message before it increases the size?

Reply:
Yes, unfortunately. The current content is unusable anyway an re-read with larger buffer does read next message instead the whole message.

Reply:
That said, I was not able to make things work without using fixed size buffer and to avoid losing the one message when buffer size increases.

Reply:
Yeah that's understandable, this buffer increase functionality would have to happen inside the net library for it to be able to consume the initial message. This may be something we need to note somewhere, I'm not really sure of the best place

Reply:
The buffer is initialized to the maximum size for a datagram socket (64k) right?  Does the other size send a single message in multiple datagrams?

Reply:
Also, since I believe this is the only conduit / pipe ... make the buffer as large as necessary from the very start should not be an issue.  i.e. buffer size = 64k

Reply:
@paramite - "Does the other si[d]e send a single message in multiple datagrams?"  Have you observed messages >64k?